genesyscloud/external_contacts/genesyscloud_externalcontacts_contact_proxy.go

```diff
@@ -0,0 +1,210 @@
+		getExternalContactIdBySearchAttr: getExternalContactIdBySearchFn,
+		updateExternalContactAttr:        updateExternalContactFn,
+	}
+}
+
+// getExternalContactsContactsProxy acts as a singleton to for the internalProxy.  It also ensures
+// that we can still proxy our tests by directly setting internalProxy package variable
+func getExternalContactsContactsProxy(clientConfig *platformclientv2.Configuration) *externalContactsContactsProxy {
+	if internalProxy == nil {
+		internalProxy = newExternalContactsContactsProxy(clientConfig)
+	}
+
+	return internalProxy
+}
+
+// getAllExternalContacts retrieves all Genesys Cloud External Contacts
+func (p *externalContactsContactsProxy) getAllExternalContacts(ctx context.Context) (*[]platformclientv2.Externalcontact, error) {
+	return p.getAllExternalContactsAttr(ctx, p)
+}
+
+// createExternalContact creates a Genesys Cloud External Contact
+func (p *externalContactsContactsProxy) createExternalContact(ctx context.Context, externalContact *platformclientv2.Externalcontact) (*platformclientv2.Externalcontact, error) {
+	return p.createExternalContactAttr(ctx, p, externalContact)
+}
+
+// DeleteExternalContact deletes a Genesys Cloud External Contact by Id
+func (p *externalContactsContactsProxy) deleteExternalContactId(ctx context.Context, externalContactId string) (int, error) {
+	return p.deleteExternalContactByIdAttr(ctx, p, externalContactId)
+}
+
+// getExternalContactById returns a single Genesys Cloud External Contact by Id
+func (p *externalContactsContactsProxy) getExternalContactById(ctx context.Context, externalContactId string) (*platformclientv2.Externalcontact, int, error) {
+	return p.getExternalContactByIdAttr(ctx, p, externalContactId)
+}
+
+// getExternalContactIdBySearch returns a single Genesys Cloud External Contact by a search term
+func (p *externalContactsContactsProxy) getExternalContactIdBySearch(ctx context.Context, search string) (externalContactId string, retryable bool, err error) {
+	return p.getExternalContactIdBySearchAttr(ctx, p, search)
+}
+
+// updateExternalContact updates a Genesys Cloud External Contact
+func (p *externalContactsContactsProxy) updateExternalContact(ctx context.Context, externalContactId string, externalContact *platformclientv2.Externalcontact) (*platformclientv2.Externalcontact, error) {
+	return p.updateExternalContactAttr(ctx, p, externalContactId, externalContact)
+}
+
+// getAllExternalContactsFn is the implementation for retrieving all external contacts in Genesys Cloud
+func getAllExternalContactsFn(ctx context.Context, p *externalContactsContactsProxy) (*[]platformclientv2.Externalcontact, error) {
+	var allExternalContacts []platformclientv2.Externalcontact
+
+	for pageNum := 1; ; pageNum++ {
+		const pageSize = 100
+
+		// The /api/v2/externalcontacts/contacts endpoint can only retrieve 10K records total.
+		// We put a constraint in make sure we never pull then 10,000 records.
+		if pageNum > 10 {
```

Comment:
This should be `> 100`, right? 
because 100 (pageSize) x 10 (pageNum) == 1_000 not 10_000